timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -35,11 +35,22 @@ def run_sigma_rule(self, query, rule_name, tag_list = None):
         """
         return_fields = []
         tagged_events_counter = 0
+        # TODO: tune the below line to be more effective
         events = self.event_stream(
             query_string=query, return_fields=return_fields)
         for event in events:
-            event.add_tags(['sigma_{0:s}'.format(rule_name)])
-            event.add_tags(tag_list)
+            ts_sigma_rule = event.source.get('ts_sigma_rule',[])
+            ts_sigma_rule.append(rule_name)
+            event.add_attributes({'ts_sigma_rule': list(set(ts_sigma_rule))})
```

Comment:
what about if the rule is already there? I know that you deal with it using the `list(set(....` but if it's already there, then there is no need to call the `add_attributes` call, that is you can save yourself an API Call if it's there already... why not

```python
if rule_name not in ts_sigma_rule:
    ts_sigma_rule.append...
    event.add_attri....
```

otherwise you just don't call the `add_attributes` and then save yourself an API call

Reply:
Yes that is a valid question, however, rules might change, esp more tags added to things, so I think "spending" the API call here is okay.